File: 0118-pascals-triangle/0118-pascals-triangle.cpp
class Solution {
public:
    vector<vector<int>> generate(int n,vector<int> curr={1},vector<vector<int>> res={}) {
        if(n==0){
           
            return res;
        }
        if(curr.size()==1){
            
            res.push_back(curr);
            curr.push_back(1);
            return generate(n-1,curr,res);
        }else{
            
            res.push_back(curr);
            vector<int> c1={1};
            for(int i=1;i<curr.size();i++){
                c1.push_back(curr[i]+curr[i-1]);
            }
            c1.push_back(1);//inserting last 1
            curr=c1;
            return generate(n-1,curr,res);        
        }
       
    }
};

File: 0118-pascals-triangle/0118-pascals-triangle.py
class Solution:
    def generate(self, n: int) -> List[List[int]]:
        res = [[1]]
        i = 1
        while i<n:
            temp = []
            for j in range (i+1):
                if(j==0 or j==i):
                    temp.append(1)
                else:
                    temp.append(res[i-1][j]+res[i-1][j-1])
            res.append(temp)
            i+=1
        return res